lib/services/sku-pack-service.js

```diff
@@ -51,6 +60,151 @@ function skuPackServiceFactory(
+    var self = this;
+    var name = uuid('v4');
+    var tmpDir = tmp();
+    var skuName = '';
+
+    return new Promise(function(resolve, reject){
+        try {
+            req.pipe(zlib.createGunzip())
+                .pipe(tar.Extract({path: tmpDir + '/' + name}))
+                .on('end', function() {
+                    return self.installPack(tmpDir + '/' + name,skuid)
+                        .spread(function(filename, contents) {
+                            skuName = path.basename(filename, '.json');
+                            return self.registerPack(filename, contents);
+                        })
+                        .then(function() {
+                            return self.regenerateSkus();
+                        }).then(function() {
+                            res.status(201).json({
+                                id: skuName
+                            });
+                        })
+                        .catch(function(e) {
+                            reject(Errors.InternalServerError(
+                                'Failed to serve file request:' + e.message));
+                        })
+                        .finally(function() {
+                            return rimrafAsync(tmpDir + '/' + name);
+                        });
+                })
+                .on('error', function() {
+                    //res.status(500).json({ error: 'Failed to serve file request' });
+                    reject(Errors.InternalServerError('Failed to serve file request'));
+                });
+        } catch(err) {
+            //res.status(500).json({ error: 'Failed to serve file request' });
+            reject(Errors.InternalServerError('Failed to serve file request'));
+        }
+    });
+
+};
+
+    SkuPackService.prototype.patchSku = function(id, body) {
+      var self = this;
+      return waterline.skus.updateByIdentifier(
+            id,
+            body
+        ).then(function (sku) {
+            return self.regenerateSkus()
+                .then(function () {
+                return sku;
+            });
+        });
+    };
+
+    SkuPackService.prototype.getNodesSkusById = function(req) {
+        return waterline.skus.needByIdentifier(req.swagger.params.identifier.values)
+            .then(function (sku) {
+            return waterline.nodes.find({ sku: sku.id });
+        });
+    };
+
+    SkuPackService.prototype.PutPackBySkuId = function(req, res) {
+        var self = this;
+        return waterline.skus.needByIdentifier(req.swagger.params.identifier.value)
+            .then(function() {
+                return self.skuPackHandler(req,res,req.swagger.params.identifier.value);
+            });
+    };
+
+    SkuPackService.prototype.DeleteSkuPackById = function(req) {
+        var self = this;
+        return waterline.skus.needByIdentifier(req.swagger.params.identifier.value)
+            .then(function() {
+                self.deletePack(req.swagger.params.identifier.value);
```

Comment:
`deletePack` will return a Promise, it must be returned, otherwise the working sequence is not as expected.